src/App.tsx

```diff
@@ -1,26 +1,159 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import { USER_ID } from './api/todos';
+import { todos as todosClient } from './api/todos';
+import { ErrorNotification } from './Components/ErrorNotification';
+import { TodoHead } from './Components/TodoHead';
+import { TodoList } from './Components/TodoList';
+import { TodoFilter } from './Components/TodoFilter';
+import { Errors } from './types/Errors';
+import { Filters } from './types/Filters';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [filter, setFilter] = useState<Filters>(Filters.all);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [inputResetFlag, setInputResetFlag] = useState(Math.random());
+  const [inputLoadingFlag, setInputLoadingFlag] = useState(false);
+  const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+  const [tempTodo, setTempTodo] = useState<string | null>(null);
+
+  const handleErrors = (error: Errors) => {
+    window.setTimeout(() => {
+      setErrorMessage(error);
+    }, 300);
+  };
+
+  // Initial TODOs loading
+  useEffect(() => {
+    todosClient
+      .get()
+      .then(res => {
+        setTodos(() => res);
+      })
+      .catch(() => {
+        handleErrors(Errors.load);
+      });
+  }, []);
+
+  //Apply Filtering
+  useEffect(() => {
+    switch (filter) {
+      case Filters.completed:
+        setFilteredTodos(todos.filter(todo => todo.completed));
+        break;
+
+      case Filters.active:
+        setFilteredTodos(todos.filter(todo => !todo.completed));
+        break;
+
+      case Filters.all:
+      default:
+        setFilteredTodos(todos);
+        break;
+    }
```

Comment:
move this logic to some function and return the filtered array from that function